platforms/software/reporting/src/main/java/org/gradle/api/reporting/internal/SimpleReport.java

```diff
@@ -28,29 +29,22 @@
 import java.io.File;
 
 public abstract class SimpleReport implements ConfigurableReport {
-    private final String name;
-    private final Describable displayName;
-    private final OutputType outputType;
 
     public SimpleReport(String name, Describable displayName, OutputType outputType) {
-        this.name = name;
-        this.displayName = displayName;
-        this.outputType = outputType;
+        this.getName().set(name);
+        this.getDisplayName().set(displayName.getDisplayName());
+        this.getOutputType().set(outputType);
     }
 
     @Override
-    public String getName() {
-        return name;
-    }
+    public abstract Property<String> getName();
 
     @Override
-    public String getDisplayName() {
-        return displayName.getDisplayName();
-    }
+    public abstract Property<String> getDisplayName();
```

Comment:
💅  Probably not that important since it's internal type, but should this be just a Provider?